### DISCUSSION TITLE
Looking for a Java client SDK for Connector API

### DISCUSSION THREAD

**Discussion by SebMarsolle** [eclipse-edc/Connector#3021] [Q&A]
Hello,
I already posted the same question in Samples https://github.com/eclipse-edc/Samples/discussions/52
I'm implementing a demonstrator application in Java,
I would like to interact with a running EDC connector to publish some assets and negotiate contracts on my "MVD",
I see in the samples that there are API we can call with simple curl commands with JSON documents,
Is there somewhere a client SDK in Java that interacts with all connector's APIs ?
Should I generate my client from a swagger or openapi yaml file ?

Regards, S. MARSOLLE from Toulouse.

**Comment by paullatzelsperger** [comment]
Hi, you can use the [connector's management API spec](https://app.swaggerhub.com/apis/eclipse-edc-bot/management-api/0.1.1-SNAPSHOT) as a starting point, but depending on the authentication you use, and because that API uses [JSON-LD](https://json-ld.org/), you will likely have to heavily modify any generated code.

**Comment by SebMarsolle** [comment]
Partially answered in https://github.com/eclipse-edc/Connector/discussions/886

I understand there is no packaged Java client.
We started to code our own dedicated client just based on the API endpoints we want to use, inspiring ourselves from JSON files from MVD and Samples.
